geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -70,30 +71,67 @@ public RedisSet(int expectedSize) {
   public RedisSet() {}
 
   public static Set<byte[]> sdiff(RegionProvider regionProvider, List<RedisKey> keys) {
-    return calculateDiff(regionProvider, keys);
+    MemberSet result = calculateDiff(regionProvider, keys, true);
+    if (result == null) {
+      return Collections.emptySet();
+    }
+    return result;
   }
 
-  private static Set<byte[]> calculateDiff(RegionProvider regionProvider, List<RedisKey> keys) {
-    RedisSet firstSet = regionProvider.getTypedRedisData(REDIS_SET, keys.get(0), true);
+  public static int sdiffstore(RegionProvider regionProvider, RedisKey destinationKey,
+      List<RedisKey> keys) {
+    MemberSet diff = calculateDiff(regionProvider, keys, false);
+    return setOpStoreResult(regionProvider, destinationKey, diff);
+  }
+
+  private static MemberSet calculateDiff(RegionProvider regionProvider, List<RedisKey> keys,
+      boolean updateStats) {
+    RedisSet firstSet = regionProvider.getTypedRedisData(REDIS_SET, keys.get(0), updateStats);
     if (firstSet.scard() == 0) {
-      return Collections.emptySet();
+      return null;
     }
-    Set<byte[]> diff = new MemberSet(firstSet.members);
+    MemberSet diff = new MemberSet(firstSet.members);
 
     for (int i = 1; i < keys.size(); i++) {
-      RedisSet curSet = regionProvider.getTypedRedisData(REDIS_SET, keys.get(i), true);
+      RedisSet curSet = regionProvider.getTypedRedisData(REDIS_SET, keys.get(i), updateStats);
       if (curSet.scard() == 0) {
         continue;
       }
 
       diff.removeAll(curSet.members);
       if (diff.isEmpty()) {
-        return Collections.emptySet();
+        return null;
       }
     }
     return diff;
   }
 
+  @VisibleForTesting
+  static int setOpStoreResult(RegionProvider regionProvider, RedisKey destinationKey,
+      MemberSet diff) {
+    RedisSet destinationSet =
+        regionProvider.getTypedRedisDataElseRemove(REDIS_SET, destinationKey, false);
+
+    if (diff == null) {
+      if (destinationSet != null) {
+        regionProvider.getDataRegion().remove(destinationKey);
+      }
+      return 0;
+    }
+
+    if (destinationSet != null) {
+      destinationSet.persistNoDelta();
+      destinationSet.members.clear();
+      destinationSet.members = diff;
+      destinationSet.storeChanges(regionProvider.getDataRegion(), destinationKey,
+          new ReplaceByteArrays(diff));
+    } else {
+      regionProvider.getDataRegion().put(destinationKey, new RedisSet(diff));
```

Comment:
since "diff" is now a MemberSet and it can be used as the members value of the RedisSet being constructed; create a new constructor on RedisSet that takes a MemberSet. It can just set "members" to diff.